lib/pc_patch.c

```diff
@@ -544,3 +544,60 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_schema(const PCPATCH *patch, const PCSCHEMA *new_schema)
+{
+	PCPATCH_UNCOMPRESSED *paout = NULL;
+	PCPOINTLIST *opl, *npl;
+	PCPOINT *opt, *npt;
+	double val;
+	size_t i, j;
+	char *name;
+	PCSCHEMA *old_schema;
+
+	old_schema = (PCSCHEMA *) patch->schema;
+
+	if ( pc_schema_similar(old_schema, new_schema) )
+	{
+		// fast path!
+		old_schema->pcid = new_schema->pcid;
+		return (PCPATCH *) patch;
+	}
+
+	// init point lists
+	opl = pc_pointlist_from_patch(patch);
+	npl = pc_pointlist_make(patch->npoints);
+
+	// build the new pointlist
+	for ( i = 0; i <patch->npoints; i++ )
+	{
+		opt = pc_pointlist_get_point(opl, i);
+		npt = pc_point_make(new_schema);
+
+		for ( j = 0; j < new_schema->ndims; j++ )
+		{
+			PCDIMENSION *odim;
+			PCDIMENSION *ndim = new_schema->dims[j];
+
+			name = ndim->name;
+			odim = pc_schema_get_dimension_by_name(old_schema, name);
+
+			val = pc_value_scale_offset(0.0, ndim);
+			if ( odim != NULL )
```

Comment:
`odim` and the default `val` could be computed only once per dimension (stored in 2 arrays rather than recomputed for each point)